### PR TITLE
CCAligner: Allow passing raw text transcript

### PR DESCRIPTION
**Pull request by HemangRajvanshy**
Added support for transcription using text files.

### CODE REVIEW

File: src/lib_ccaligner/grammar_tools.cpp [hunk 1]
```diff
@@ -17,6 +17,263 @@ static int systemGetStatus(const char* command) {
...
+		std::istringstream iss(line);
+		allData += line + "\n";
+	}
+	return allData;
+}
+
+bool generate(std::string transcriptFileName, grammarName name)
+{
+	std::string transcript = getFileData(transcriptFileName);
+	std::cout << transcript << "\n";
```
Comment (saurabhshri) [inline src/lib_ccaligner/grammar_tools.cpp]: This looks like remains of some debugging :)

File: src/lib_ccaligner/grammar_tools.cpp [hunk 2]
```diff
@@ -17,6 +17,263 @@ static int systemGetStatus(const char* command) {
 #endif
 }
 
+void PerformChecks(grammarName name, bool generateQuickLM)
```
Comment (saurabhshri) [inline src/lib_ccaligner/grammar_tools.cpp]: The name really doesn't match to what's being done.

**Comment by saurabhshri** [comment]
Also, the two generate functions share a lot of common operations, so it makes sense to remove the redundancy and only have that code once. This would also help clean up the code.

Really good job with the PR. With the first look it appears good. Please make the changes suggested .

File: src/lib_ccaligner/grammar_tools.cpp [hunk 3]
```diff
@@ -17,9 +17,274 @@ static int systemGetStatus(const char* command) {
...
+			phoneticCorpusDump.open("tempFiles/corpus/phoneticCorpus.txt", std::ios::binary);
+			phoneticCorpusDump.close();
+		}
+
+		catch (std::system_error& e)
+		{
+			FATAL(EXIT_FAILURE, e.code().message().c_str());
+		}
+
+	}
```
Comment (saurabhshri) [inline src/lib_ccaligner/grammar_tools.cpp]: The lines from 146 till here also looks like the same shared between the two generate functions. Can you look at them as well?

Comment (HemangRajvanshy) [reply src/lib_ccaligner/grammar_tools.cpp]: Sure. Will do asap.

File: src/lib_ccaligner/grammar_tools.cpp [hunk 4]
```diff
@@ -17,6 +17,92 @@ static int systemGetStatus(const char* command) {
 #endif
 }
 
+void InitializeGenerator(bool &generateQuickDict, bool &generateQuickLM, grammarName &name)
```
Comment (saurabhshri) [inline src/lib_ccaligner/grammar_tools.cpp]: Please rename it to something more appropriate like `ConfigureQuickGenerationOptions`.

File: src/lib_ccaligner/grammar_tools.cpp [hunk 5]
```diff
@@ -17,6 +17,92 @@ static int systemGetStatus(const char* command) {
...
+
+	if (rv != 0)
+	{
+		FATAL(EXIT_FAILURE, "Unable to create directory tempFiles/ : %s", strerror(errno));
+	}
+
+	LOG("Directories created successfully!");
+}
+
+void InitializeDumpFiles(grammarName name, std::ofstream &corpusDump, std::ofstream &fsgDump,
```
Comment (saurabhshri) [inline src/lib_ccaligner/grammar_tools.cpp]: Please rename it to something more appropriate like `CreateNewGrammarFiles`.

**Pull request merged**